clients/include/type_dispatch.hpp

```diff
@@ -41,10 +41,19 @@ auto rocblas_simple_dispatch(const Arguments& arg)
 template <template <typename...> class TEST>
 auto rocblas_blas1_dispatch(const Arguments& arg)
 {
-    const auto Ti = arg.a_type, To = arg.d_type;
-
+    const auto Ti = arg.a_type, Tb = arg.b_type, To = arg.d_type;
     if(Ti == To)
-        return rocblas_simple_dispatch<TEST>(arg);
+    {
+        if(Tb == Ti)
+            return rocblas_simple_dispatch<TEST>(arg);
+        else
+        { // for csscal and zdscal only
+            if(Ti == rocblas_datatype_f32_c && Tb == rocblas_datatype_f32_r)
```

Comment:
The types should be changed here, i.e.:
`if(Ti == rocblas_datatype_f32_r && Tb == rocblas_datatype_f32_c)`
    `return TEST<float, rocblas_float_complex>{}(arg)`
and the same with double precision types.